### PR TITLE
Allow configuring the application's config path

### PR DESCRIPTION
**Pull request by lewisd32**
Allow configuring the application's config path through system property or method overriding.

eg.
```Java
private final ApplicationContext<TestConfig> appContext = new ApplicationContext<TestConfig>() {
        @Override
        protected Class<? extends Application<TestConfig>> mainClass() {
            return TestApp.class;
        }

        @Override
        protected String defaultConfigPath() {
            return "path-to-config.yml";
        }

    };
```

### CODE REVIEW

File: unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/OneShotApplicationState.java [hunk 1]
```diff
@@ -26,6 +28,10 @@ private String appUrlFromPort(int localPort) {
         return String.format("http://localhost:%d/", localPort);
     }
 
+    public String configPath() {
```
Comment (ojacobson) [inline unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/OneShotApplicationState.java]: This should probably be pure abstract, like the other helpers from `ApplicationContext`. If we're merging #28 instead, though, eh.

File: unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/OneShotApplicationState.java [hunk 2]
```diff
@@ -6,6 +6,8 @@
 import io.dropwizard.testing.junit.DropwizardAppRule;
 import io.github.unacceptable.lazy.Lazily;
 
+import static io.github.unacceptable.lazy.Lazily.systemProperty;
```
Comment (ojacobson) [inline unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/OneShotApplicationState.java]: ?

File: unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java [hunk 3]
```diff
@@ -111,4 +96,46 @@ public void detectsExistingApp() {
...
+        ApplicationContext<TestConfig> applicationContext = new ApplicationContext<TestConfig>() {
+            @Override
+            protected Class<? extends Application<TestConfig>> mainClass() {
+                return TestApp.class;
+            }
+
+            @Override
+            protected String defaultConfigPath() {
+                File workingDir = new File(".");
+                File configFile = new File(workingDir, "src/test/resources/test-app-on-slash-api.yml");
```
Comment (ojacobson) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java]: This is strictly equivalent to

```java
File configFile = new File("src/test/resources/test-app-on-slash-api.yml");
```

**Comment by ojacobson** [comment]
I like it. I don't use that feature, but I bet a non-trivial number of Dropwizard users would.

File: unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/OneShotApplicationState.java (hunk 2 above)
Comment (lewisd32) [inline unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/OneShotApplicationState.java]: I... have no idea.

File: unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java (hunk 3 above)
Comment (lewisd32) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java]: Yes... of course it is.  That's blindingly obvious, I have no idea how I managed to write that without realizing. :joy: I blame too much flying.

**Comment by lewisd32** [comment]
Is this entirely superseded by #28? I feel like it probably is.

**Comment by ojacobson** [comment]
It is.

**Comment by lewisd32** [comment]
Superseded by #28.

**Pull request closed without merging**